api/catalog/management/commands/generatewaveforms.py

```diff
@@ -0,0 +1,127 @@
+import logging
+import subprocess
+
+from catalog.api.models.audio import Audio, AudioAddOn
+from django_tqdm import BaseCommand
+from limit import limit
+
+
+def paginate_reducing_query(get_query_set, page_size=10):
+    """
+    We can't use `Paginator` because it can't handle the situation
+    where the query result changes each time a page is accessed.
+    Because the `audios` QuerySet result is naturally getting smaller
+    each time we successfully process waveforms, we can just take
+    the first ten for each "page" until the page comes back empty.
+    This should theoretically be faster/less DB latency inducing
+    anyway as we're never going to have huge OFFSET values to
+    access deep pages.
+    """
+    page = list(get_query_set()[0:page_size])
+    while len(page):
+        yield page
+        page = list(get_query_set()[0:page_size])
+
+
+class Command(BaseCommand):
+    help = "Generates waveforms for all audio records to populate the cache."
+    """
+    Note: We rely on the file download and waveform generation times
+    taking long enough to prevent us from either making too many requests
+    to the upstream provider or inserting into our database too quickly and
+    causing a slow down. In local tests and in tests run on the staging server
+    it appeared to take on average around 6 to 8 seconds for each audio file.
+    That should be enough latency to not cause any problems.
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--unlimited", help="Remove self impose rate limits for testing."
+        )
+        parser.add_argument(
+            "--limit", help="Limit the number of waveforms to create.", type=int
```

Comment:
Just a thought, the fact that these both have `limit` in them is kind of confusing. Maybe we could do:
* `--unlimited` -> `--no-rate-limit`
* `--limit` -> `--max-records`

What do you think?

Reply:
Yes, I like these better. I hated the ones I chose!

Reply:
Done in e75964f68a12029b70c8c950945dcc0a9f907a95